code/controllers/subsystem/weeds.dm

```diff
@@ -56,6 +56,8 @@ SUBSYSTEM_DEF(weeds)
 			return
 		// Adds a bit of jitter to the spawning weeds.
 		addtimer(CALLBACK(src, .proc/create_weed, T, creating[T]), rand(1, 3 SECONDS))
+		if(prob(20) && !(locate(/obj/alien/weeds/node) in range(1, T)))
+			addtimer(CALLBACK(src, .proc/create_node, T, creating[T]), rand(8, 10 SECONDS))
```

Comment:
You are creating a weed patch then eventually a node. You don't need to create a weed patch if you create a node

Reply:
I think i do, because not creating a weed patch would leave walls empty

Reply:
No. Check if that place can hold a node, roll a dice, create a node timer. If one of these conditions is false, create a weed timer